algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/workload.py

```diff
@@ -0,0 +1,253 @@
+from algorithmic_efficiency.workloads.librispeech_conformer import metrics
+from algorithmic_efficiency.workloads.librispeech_conformer import workload
+
+from .model import ConformerConfig
+from .model import ConformerEncoderDecoder
+from .model import initialize
+
+USE_PYTORCH_DDP, RANK, DEVICE, N_GPUS = pytorch_setup()
+
+
+class LibriSpeechConformerWorkload(workload.BaseLibrispeechWorkload):
+
+  def init_model_fn(self, rng: spec.RandomState) -> spec.ModelInitState:
+    torch.random.manual_seed(rng[0])
+    model = ConformerEncoderDecoder(ConformerConfig())
+    self._model = model
+    self.ctc_loss = torch.nn.CTCLoss(blank=0, reduction="none")
+    # Run model once to initialize lazy layers
+    t = 320000
+    wave = torch.randn((2, t))
+    pad = torch.zeros_like(wave)
+    _ = model(wave, pad)
+    initialize(model)
+
+    self._param_shapes = {
+        k: spec.ShapeTuple(v.shape) for k, v in model.named_parameters()
+    }
+    model.to(DEVICE)
+    if N_GPUS > 1:
+      if USE_PYTORCH_DDP:
+        model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+        model = DDP(model, device_ids=[RANK], output_device=RANK)
+      else:
+        model = torch.nn.DataParallel(model)
+    return model, None
+
+  def init_metrics_bundle(self, tokenizer_vocab_path):
+    logging.info('Initializing metrics bundle.')
+    self.tokenizer = metrics.load_tokenizer(tokenizer_vocab_path)
+
+  def model_fn(
+      self,
+      params: spec.ParameterContainer,
+      augmented_and_preprocessed_input_batch: Dict[str, spec.Tensor],
+      model_state: spec.ModelAuxiliaryState,
+      mode: spec.ForwardPassMode,
+      rng: spec.RandomState,
+      update_batch_norm: bool) -> Tuple[spec.Tensor, spec.ModelAuxiliaryState]:
+    del model_state
+    del rng
+    del update_batch_norm
+
+    model = params
+
+    if mode == spec.ForwardPassMode.EVAL:
+      model.eval()
+
+    if mode == spec.ForwardPassMode.TRAIN:
+      model.train()
+
+    contexts = {
+        spec.ForwardPassMode.EVAL: torch.no_grad,
+        spec.ForwardPassMode.TRAIN: contextlib.nullcontext
+    }
+
+    with contexts[mode]():
+      logits, logits_paddings = model(
+        augmented_and_preprocessed_input_batch['inputs'],
+        augmented_and_preprocessed_input_batch['input_paddings'])
+
+    return (logits, logits_paddings), None
+
+  @property
+  def model_params_types(self):
+    if self._param_types is None:
+      self._param_types = param_utils.pytorch_param_types(self._param_shapes)
+    return self._param_types
+
+  def build_input_queue(self,
+                        data_rng: jax.random.PRNGKey,
+                        split: str,
+                        data_dir: str,
+                        global_batch_size: int,
+                        num_batches: Optional[int] = None,
+                        repeat_final_dataset: bool = False):
+    per_device_batch_size = int(global_batch_size / N_GPUS)
+    n_inputs = 4
```

Comment:
nit: delete this and below use `enumerate(keys)`

Reply:
Done.